week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,92 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+//Question 1
+// Find retorna só o primeiro que satisfaz a condiçao. Filter retorna mais!
+
+function checkingDistance(travelDestinations) {
+  if (travelDestinations.distanceKms <= 500) {
+    return travelDestinations;
+  }
+}
+function getName(travelDestinations) {
+  return travelDestinations.destinationName;
+}
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(checkingDistance)
+  .map(getName);
+
+//Question 2
 
+function isItReachableByFerry(travelDestinations) {
+  return travelDestinations.transportations.includes("ferry");
+}
 
-var destinationNamesWithin500Kms = // Complete here
+var destinationNameReachableByFerry = travelDestinations
+  .filter(isItReachableByFerry)
+  .map(getName);
 
-var destinationNameReachableByFerry = // Complete here
+//Question 3
 
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+function destinationMoreThan300andByTrain(travelDestinations) {
+  return (
+    travelDestinations.transportations.includes("train") &&
+    travelDestinations.distanceKms > 300
+  );
+}
 
+var destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(destinationMoreThan300andByTrain)
+  .map(getName);
 
```

Comment:
Good.  

Minor point: again, your predicate function takes one parameter, so better to call it `destination`.  After all, the statement `travelDestinations.distanceKms > 300` makes no sense if travelDestinations is really multiple.